### PR TITLE
get order status by hash

### PR DESCRIPTION
**Pull request by frolovdev**
getOrderStatus

### CODE REVIEW

File: src/api/orders/types.ts [hunk 1]
```diff
@@ -22,3 +23,36 @@ export type ActiveOrder = {
...
+export type OrderStatusResponse = {
+    status: OrderStatusMapped
+    order: LimitOrderV3Struct
+    points: AuctionPoint[] | null
+    fills: Fill[]
+    auctionStartDate: number
+    auctionDuration: number
+    initialRateBump: number
+    isNativeCurrency: boolean
+}
```
Comment (frolovdev) [inline src/api/orders/types.ts:58]: 📓 I don't sure about the types. Please check it

File: src/api/orders/orders.request.ts [hunk 2]
```diff
@@ -21,3 +21,21 @@ export class ActiveOrdersRequest {
...
+    static new(params: OrderStatusParams): OrderStatusRequest {
+        return new OrderStatusRequest(params)
+    }
+
+    build(): OrderStatusParams {
+        return {
+            orderHash: this.orderHash
+        }
+    }
+}
```
Comment (krboktv) [inline src/api/orders/orders.request.ts:41]: Add `validation` function like here:
https://github.com/1inch/fusion-sdk/blob/20679a15cea5b49bd0f86b5fb237726d80d09b01/src/api/quoter/quoter.request.ts#L32

Check that `orderHash.length === 66` and the string is hexdecimal

File: src/api/orders/types.ts [hunk 3]
```diff
@@ -22,3 +23,36 @@ export type ActiveOrder = {
...
+    PartiallyFilled = 'partially-filled',
+    WrongPermit = 'wrong-permit',
+    Cancelled = 'cancelled',
+    InvalidSignature = 'invalid-signature'
+}
+
+export type Fill = {
+    txHash: string | null
+    filledMakerAmount: string | null
+    filledAuctionTakerAmount: string | null
```
Comment (krboktv) [inline src/api/orders/types.ts:46]: remove `null` here

File: src/api/orders/types.ts [hunk 4]
```diff
@@ -22,3 +23,36 @@ export type ActiveOrder = {
 }
 
 export type ActiveOrdersResponse = PaginationOutput<ActiveOrder>
+
+export type OrderStatusParams = {
+    orderHash: string
+}
+
+export enum OrderStatusMapped {
```
Comment (krboktv) [inline src/api/orders/types.ts:31]: Call it `OrderStatus`

File: src/sdk/sdk.ts [hunk 5]
```diff
@@ -33,6 +36,14 @@ export class FusionSDK {
         return this.api.getActiveOrders(request)
     }
 
+    async getOrderStatus({
+        orderHash
+    }: OrderStatusParams): Promise<OrderStatusResponse> {
```
Comment (krboktv) [inline src/sdk/sdk.ts:41]: Here you can use just `orderHash: string` as a parameter

**Review by krboktv — CHANGES_REQUESTED** [review]

File: src/sdk/sdk.ts (hunk 5 above)
Comment (frolovdev) [reply src/sdk/sdk.ts:41]: If we decide to change the API, we will need to break the contract 🤔  by refactoring back to the object

File: src/api/orders/orders.request.ts (hunk 2 above)
Comment (frolovdev) [reply src/api/orders/orders.request.ts]: done

File: src/sdk/sdk.ts (hunk 5 above)
Comment (krboktv) [reply src/sdk/sdk.ts:41]: I don't think this may changed in the future

**Review by krboktv — APPROVED** [review]

**Pull request merged**